src/main/java/org/ilastik/ilastik4ij/hdf5/Hdf5DataSetWriterFromImgPlus.java

```
@@ -39,400 +39,384 @@
+            }
+
+            try {
+                if (dataspace_id >= 0) {
+                    H5Sclose(dataspace_id);
+                    dataspace_id = -1;
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            try {
+                H5Pclose(dcpl_id);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            try {
+                if (dataset_id >= 0)
+                    H5Dclose(dataset_id);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            try {
+                if (file_id >= 0)
+                    H5Fclose(file_id);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        } catch (HDF5Exception err) {
+            log.error("Error while saving '" + filename + "':\n" + err);
+        } catch (Exception err) {
+            log.error("Error while saving '" + filename + "':\n" + err);
+        } catch (OutOfMemoryError o) {
+            log.error("Out of Memory Error while saving '" + filename + "':\n" + o);
+        }
+    }
+
+    private void writeARGB() {
+        long[] channelDimsRGB = new long[5];
+        channelDimsRGB[0] = nFrames; //t
+        channelDimsRGB[1] = nLevs; //z
+        channelDimsRGB[2] = nRows; //y
+        channelDimsRGB[3] = nCols; //x
+        channelDimsRGB[4] = NUM_OF_ARGB_CHANNELS;//c
+
+        long[] colorIniDims = new long[5];
+        colorIniDims[0] = 1;
+        colorIniDims[1] = 1;
+        colorIniDims[2] = nRows;
+        colorIniDims[3] = nCols;
+        colorIniDims[4] = 1;
+
+        try {
+            dataspace_id = H5.H5Screate_simple(5, colorIniDims, maxdims);
+        } catch (Exception e) {
+            log.error("H5D dataspace creation failed", e);
+            throw e;
+        }
+
+        try {
+            if ((file_id >= 0) && (dataspace_id >= 0))
```

btw, I'd expect `H5.H5Screate_simple(...)` to throw an exception instead of returning `dataset_id` equal to -1. Java's not C :smile: so please get rid of those `if(dataspace_id >= 0)` and other such conditions. Have a look at h5 docs to double check

---

Changes scheduled in next PR which basically deals with refactoring everything in the *Hdf5DataSetWriterFromImgPlus* class.